### ISSUE TITLE
`os.path.basename` on windows gives empty strings

### ISSUE THREAD

**Issue by ewerybody** [python/cpython#104093]
# Bug report

I wanted to check perforce stream names like `//depot_name/stream_name` by passing them into `os.path.basename` to get `stream_name` But I ended up getting **empty strings**!

```
Python 3.12.0a7 (tags/v3.12.0a7:b861ba4, Apr  4 2023, 16:33:41) [MSC v.1934 64 bit (AMD64)] on win32
>>> import os
>>> os.path.basename('//sdfgsd')
''
>>> os.path.basename('//sdfgsd/asd')
''
>>> os.path.basename('//sdfgsd/asd/fsd')
'fsd'
>>>
```

# Your environment

- CPython versions tested on: **Python 3.12.0a7** 
- Operating system and architecture: **Windows 10 Pro 64-bit 22H2 19045.2846**

## versions tested

It works right away **as expected** on the very Python homepage and on my Android phone the same way
```
Python 3.10.5 (main, Jul 22 2022, 17:09:35) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.path.basename('//sdfsdfg')
'sdfsdfg'
>>> os.path.basename('//sdfsdfg/asd')
'asd'
```

Weird enough the behaviour is a little different on `3.10.5`:
```
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.path.basename('//asdfsdf')
'asdfsdf'
>>> os.path.basename('//asdfsdf/asdf')
''
>>> os.path.basename('//asdfsdf/asdf/sdc')
'sdc'
```
So in any environment it works with at least 3 parts! On windows `3.10.5` works with 1 part, but not with 2, all later versions from `3.11` yield empty strings with 1 and 2 parts.

I guess this is due to `\\` being the default separator `os.sep` on Windows? And I should split with `os.altsep`?
But still it's not breaking consistently. Any more input?
Thanks :)

**Comment by eryksun** [comment]
"//depot_name/stream_name" is a UNC path, where "depot_name" is the server name and "stream_name" is the share name. A UNC share is considered the drive part of a Windows path. For example, if the current working directory is "\\\\depot_name\\stream_name\\spam", then Windows resolves the relative path "\\eggs" as the absolute path "\\\\depot_name\\stream_name\\eggs".

Support for UNC paths was fixed in 3.11. A path that starts with two or more path separators (backslash or slash) is now always handled correctly as a UNC path, even if it's incomplete with an empty server or empty share component. For example:

3.10:
```pycon
>>> os.path.splitdrive('//server')
('', '//server')
```

3.11:
```pycon
>>> os.path.splitdrive('//server')
('//server', '')
```

**Comment by ericvsmith** [comment]
I'm going to close this, since it looks like it's already been fixed in 3.11.
